model-optimizer/mo/moc_frontend/extractor.py

```diff
@@ -1,35 +1,70 @@
 # Copyright (C) 2018-2021 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-import logging as log
 import re
-from collections import defaultdict
-from copy import copy
-
-import numpy as np
 
+from mo.front.extractor import raise_no_node, raise_node_name_collision
 from mo.utils.error import Error
 
-from ngraph.frontend import InputModel  # pylint: disable=no-name-in-module,import-error
+from ngraph.frontend import InputModel, Place  # pylint: disable=no-name-in-module,import-error
+
+import numpy as np
 
 
 def decode_name_with_port(input_model: InputModel, node_name: str):
     """
     Decode name with optional port specification w/o traversing all the nodes in the graph
-    TODO: in future node_name can specify input/output port groups and indices (58562)
+    TODO: in future node_name can specify input/output port groups as well as indices (58562)
     :param input_model: Input Model
     :param node_name: user provided node name
     :return: decoded place in the graph
     """
-    # Check exact match with one of the names in the graph first
+    found_nodes = []
+    found_node_names = []
+
     node = input_model.get_place_by_tensor_name(node_name)
-    if node:
-        return node
+    found_node_names.append('Tensor:' + node_name) if node else None
+    found_nodes.append(node) if node else None
+
+    node = input_model.get_place_by_operation_name(node_name)
+    found_node_names.append('Operation:' + node_name) if node else None
+    found_nodes.append(node) if node else None
+
+    regexp_post = r'(.+):(\d+)'
+    match_post = re.search(regexp_post, node_name)
+    if match_post:
+        node_post = input_model.get_place_by_operation_name(match_post.group(1))
+        node_post = node_post.get_output_port(
+            outputPortIndex=int(match_post.group(2))) if node_post else None
+        found_node_names.append(match_post.group(1)) if node_post else None
+        found_nodes.append(node_post) if node_post else None
```

Comment:
The same comment again. And the same for lines 46-49

Reply:
Fixed